pkg/filters/filter_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+package filters
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"storj.io/storj/pkg/storj"
+)
+
+var pieceIDs [][]byte
+var initDone bool
+var nbPiecesInFilter int
+var totalNbPieces int
+var falsePositiveProbability float64
+
+//  generates 1 million piece ids
+// adds 95% of them to the bloom filter,
+// and then checks all 1 million piece ids with the bloom filter
+// measure times, memory allocation, false positives
+
+func Init() {
+	if !initDone {
+		totalNbPieces = 1000000
+		nbPiecesInFilter = 950000
+		pieceIDs = GenerateIDs(totalNbPieces)
+		initDone = true
+		falsePositiveProbability = 0.1
+	}
+}
+
+func benchmarkAdd(filter Filter, pieceIDs [][]byte, b *testing.B) {
+	for _, pieceID := range pieceIDs {
+		filter.Add(pieceID)
+	}
+}
+
+func benchmarkContains(filter Filter, pieceIDs [][]byte, b *testing.B) (nbPiecesIn int) {
+	nbPiecesIn = 0
+	for _, pieceID := range pieceIDs {
+		if filter.Contains(pieceID) {
+			nbPiecesIn++
+		}
+	}
+	return
+}
+
+func benchmarkFilter(filter Filter, pieceIDs [][]byte, b *testing.B) (p float64) {
+	b.ReportAllocs()
```

Comment:
I think that this should be controlled by the proper Bench function rather than the helper one.